### PR TITLE
chat attachments: use SimpleFS for KBFS paths; also ensure Close()

### PR DESCRIPTION
**Pull request by songgao**
Will use this for sending attachments to chat from Files tab.

### CODE REVIEW

File: go/chat/attachments/utils.go [hunk 1]
```diff
@@ -120,6 +129,85 @@ func (f *FileReadResetter) Close() error {
...
+type KbfsReadResetter struct {
+	client *keybase1.SimpleFSClient
+	opid   keybase1.OpID
+	offset int64
+	ctx    context.Context
+}
+
+const (
+	kbfsPrefix        = "/keybase"
+	kbfsPrefixPrivate = "/keybase/private/"
```
Comment (strib) [inline go/chat/attachments/utils.go]: Use `kbfsPrefix+` to define the rest of these constants?

File: go/chat/attachments/sender.go [hunk 2]
```diff
@@ -27,7 +27,12 @@ func NewSender(gc *globals.Context) *Sender {
 
 func (s *Sender) MakePreview(ctx context.Context, filename string, outboxID chat1.OutboxID) (res chat1.MakePreviewRes, err error) {
 	defer s.Trace(ctx, func() error { return err }, "MakePreview")()
-	src, err := NewFileReadResetter(filename)
+	var src ReadCloseResetter
+	if IsKbfsPath(filename) {
```
Comment (mmaxim) [inline go/chat/attachments/sender.go]: Would be nice if this got moved out to some other function.

**Comment by songgao** [comment]
ready for another look thanks!

**Comment by songgao** [comment]
Thanks for looking! I just tried this in sim with network link conditioner on 3G; tried backgrouding and killing app and uploading still worked. Seems OK. Merging!

**Pull request merged**